### PR TITLE
Lra 989 room ready add archive param to reports

### PR DESCRIPTION
**Pull request by Mohammad4844**
- added an 'archive' param to all reports
  - they filter to include ONLY archived options
  - special cases: for reports like common_attribute_states_report, it will show only records that have BOTH rooms and common attributes archived

### CODE REVIEW

**Comment by britaumich** [comment]
@the inspection_rate_report includes rooms that were never checked if Show Archived Records?  is checked. To test add a new building to a zone, but don't check any rooms in it.
It looks like the other reports work fine.

**Comment by britaumich** [comment]
@Mohammad4844
I think `.where(archived: archived)` should be included in this query (line 132) too:
```
rooms_no_room_state = Room.active
                                .left_outer_joins(:room_states)
                                .joins(floor: { building: :zone })
                                .where(buildings: { id: building_id, zone_id: zone_id })
                                .where(room_states: { id: nil })
                                .where.not(id: rooms.map(&:id))
```

**Comment by Mohammad4844** [comment]
> @the inspection_rate_report includes rooms that were never checked if Show Archived Records? is checked. To test add a new building to a zone, but don't check any rooms in it. It looks like the other reports work fine.

@britaumich  Im not quite sure how to recreate this (or maybe I just don't understand it). But I think the missing `.where(archived: archived)` I just added should fix it. Can you check and let me know

File: app/controllers/reports_controller.rb [hunk 1]
```diff
@@ -118,20 +118,21 @@ def inspection_rate_report
...
                   .group('rooms.id')
                   .select('rooms.*')
                   .select('COUNT(room_states.id) AS room_check_count')
                   .order('room_check_count DESC')
 
-      rooms_no_room_state = Room.active
-                                .left_outer_joins(:room_states)
-                                .joins(floor: { building: :zone })
+      rooms_no_room_state = Room.left_outer_joins(floor: { building: :zone })
+                                .joins(:room_states)
```
Comment (britaumich) [inline app/controllers/reports_controller.rb:134]: For the inspection_rate_report rooms_no_room_state query has rooms that were never checked (have no room_state; this is why there is a separate query for these rooms). In the report's result, these rooms have 0.0% Inspection Rate. These rooms would never be archived so we don't need to run this query for archived data. And with `.where(archived: archived)` the query returns zero records even if there are rooms without room_states.

I suggest to do the following for this report

```
if archived
     rooms_no_room_state = []
else
     rooms_no_room_state = Room.active
                                .left_outer_joins(:room_states)
                                .joins(floor: { building: :zone })
                                .where(buildings: { id: building_id, zone_id: zone_id })
                                .where(room_states: { id: nil })
                                .where.not(id: rooms.map(&:id))
                                .select('rooms.*')
                                .select('0 AS room_check_count')
                                .order('rooms.room_number')
 end
```

**Review by britaumich — COMMENTED** [review]
Also, could you please sort @data for the inspection_rate_report:
`@data = @data.sort_by { |k| [k[4], k[2], k[0]] }.reverse!`

**Review by britaumich — APPROVED** [review]

**Pull request merged**